src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -664,7 +665,7 @@ Literal Literal::as_language_tag_eq(Literal const &other, Node::NodeStorage &nod
     }                                                  \
     RDF4CPP_DETAIL_TRY_WRITE_STR("\"");
 
-bool Literal::serialize(void *const buffer, writer::Cursor *cursor, writer::FlushFunc const flush) const noexcept {
+bool Literal::serialize(void *const buffer, writer::Cursor *cursor, writer::FlushFunc const flush, const writer::SerializationState* state) const noexcept {
```

Comment:
Please create two versions of this function. (They can share the same impl switched by a bool)

- keep `Literal::serialize` interface as it was
- add `Literal::serialize_short` (better name would be good) that serializes for formats like turtle, i.e. shortens the datatype IRI 

Note that datatype shortening can be done without any additional information from the outside for fixed id datatypes (which is what we want).

Have a look at `datatypes::registry::reserved_datatype_ids` you can (at compiletime) create a `array<string>` (index by LiteralType) out of it that stores the shortened urls for those datatypes.

You can get the literal type to index the array out of the backend handle

Reply:
*from the outside meaning without an additional function param